pkg/engine/renderer/enhancer_test.go

```diff
@@ -141,6 +141,86 @@ func TestEnhancerApply_noAdditionalMetadata(t *testing.T) {
 	}
 }
 
+func TestEnhancerApply_dependencyHash(t *testing.T) {
+
+	ss := statefulSet("statefulset", "default")
+
+	ss.Spec.Template.Spec.Volumes = append(ss.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: "configMap",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: "configmap"},
+			},
+		},
+	})
+
+	tpls := map[string]string{
+		"statefulset": resourceAsString(ss),
+		"configmap":   resourceAsString(configMap("configmap", "default")),
+	}
+
+	meta := metadata()
+	meta.PlanUID = uuid.NewUUID()
+
+	e := &DefaultEnhancer{
+		Scheme:    utils.Scheme(),
+		Discovery: fake.CachedDiscoveryClient(),
+	}
+
+	objs, err := e.Apply(tpls, meta)
+	if err != nil {
+		t.Errorf("failed to apply template %s", err)
+	}
+
+	for _, o := range objs {
+		unstructMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(o)
+		if err != nil {
+			t.Errorf("failed to parse object to unstructured: %s", err)
+		}
+		if o.(metav1.Object).GetName() != "statefulset" {
+			continue
+		}
+
+		f, ok, _ := unstructured.NestedFieldNoCopy(unstructMap, "spec", "template", "metadata", "annotations")
+		assert.NotNil(t, f)
+		assert.True(t, ok, "Statefulset contains no annotations")
+
+		annotations, _ := f.(map[string]interface{})
+
+		hash, ok := annotations[kudo.DependenciesHashAnnotation]
+		assert.NotNil(t, hash)
+		assert.Equal(t, "929a2dffa86ad2460fdcf72977998bd0", hash, "Hashes are not the same")
+		assert.True(t, ok, "Statefulset contains no dependency hash field")
+	}
+}
```

Comment:
I'd like to see more test coverage on the unit test level. For example, could you add a test, where a parameter change leads to a hash **change** (where a hash is already set)? And could you also through in an independent object (e.g. a Deployment) and check that the hash is not set on it?

Reply:
Added unit tests, let me know if you have another idea for one.